src/main/java/org/broadinstitute/hellbender/tools/copynumber/CollectFragmentCounts.java

```diff
@@ -0,0 +1,223 @@
+
+        final List<SimpleInterval> intervals = intervalArgumentCollection.getIntervals(sequenceDictionary);
+        cachedIntervalList = new CachedBinarySearchIntervalList<>(intervals);
+        intervalCounts = new int[intervals.size()];
+
+        logger.info("Starting to collect fragment counts...");
+    }
+
+    @Override
+    public void apply(GATKRead read, ReferenceContext referenceContext, FeatureContext featureContext) {
+        //getting a center of the fragment
+
+        //TODO collect information on reads that do not have a properly paired mate
+        int centerOfFragment = ReadOrientation.getCenterOfFragment(read);
+        final Locatable centerFragmentLocation = new SimpleInterval(read.getContig(), centerOfFragment, centerOfFragment);
+
+        final IndexRange intersectionRange = cachedIntervalList.findIntersectionRange(centerFragmentLocation);
+        if (intersectionRange.size() > 1) {
+            // should not reach here since intervals are checked for overlapping;
+            // doing a check to protect against future code changes
+            throw new GATKException.ShouldNeverReachHereException("At most one interval can intersect with a center of a fragment.");
+        }
+        intersectionRange.forEach(intervalIndex -> intervalCounts[intervalIndex]++);
+    }
+
+    @Override
+    public Object onTraversalSuccess() {
+        logger.info("Finished collecting fragment files");
+        logger.info("Writing results to disk");
+        writeResults(outputCountsFile, sampleName);
+
+        logger.info("Fragment counts written to " + outputCountsFile.toString());
+        return "SUCCESS";
+    }
+
+    private void writeResults(final File outputFile, final String sampleName) {
+        //TODO replace this with the TableWriter
+        try (PrintWriter countsFileWriter = new PrintWriter(outputFile)) {
+            countsFileWriter.println(composeHeader(getCommandLine(), sampleName));
+            for (int i = 0; i < intervalCounts.length; i++) {
+                final SimpleInterval nextInterval = cachedIntervalList.getSortedIntervals().get(i);
+                //creating the target just to get a properly formatted name that matches other current tools
+                final Target targetFromInterval = new Target(nextInterval);
+                String nextIntervalCountLine = String.join(COLUMN_SEPARATOR,
+                        nextInterval.getContig(),
+                        Integer.toString(nextInterval.getStart()),
+                        Integer.toString(nextInterval.getEnd()),
+                        targetFromInterval.getName(),
+                        Integer.toString(intervalCounts[i]));
+                countsFileWriter.println(nextIntervalCountLine);
+            }
+        } catch (IOException exc) {
+            throw new UserException.CouldNotCreateOutputFile(outputFile, exc);
+        }
+    }
+
+    private static String composeHeader(final String commandLine, final String sampleName) {
```

Comment:
Same thing with this style of header.  The sooner we can move towards consistent file formats, the better.